x-pack/plugins/security_solution/public/flyout/left/components/correlations_details_alerts_table.tsx

```diff
@@ -110,11 +126,35 @@ export const CorrelationsDetailsAlertsTable: FC<CorrelationsDetailsAlertsTablePr
       );
   }, [data]);
 
+  const shouldUseFilters = Boolean(
+    alertIds && alertIds.length && alertIds.length >= dataProviderLimit
+  );
+  const dataProviders = useMemo(
+    () => (shouldUseFilters ? null : getDataProviders(scopeId, eventId, alertIds)),
+    [alertIds, shouldUseFilters, scopeId, eventId]
+  );
+  const filters: Filter[] | null = useMemo(
+    () => (shouldUseFilters ? getFilters(alertIds) : null),
+    [alertIds, shouldUseFilters]
+  );
+
   return (
     <ExpandablePanel
       header={{
         title,
         iconType: 'warning',
+        headerContent: (
+          <div data-test-subj={INVESTIGATE_IN_TIMELINE_BUTTON_TEST_ID}>
```

Comment:
I made a small change here where I'm using `dataTestSubj` instead of a static value. This will be useful when doing Cypress tests. Having a static value would mean multiple hits when fetching by id. Having dynamic values will allow us to target just same source event, or just ancestry investigate in timeline button...